specs/CustomDownload.md

```diff
@@ -0,0 +1,784 @@
+
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                // This function can be called to show the download dialog and
+                // complete the event at a later time, allowing the developer to
+                // perform async work before the event completes.
+                m_completeDeferredDownloadEvent = [showDialog, deferral] {
+                    showDialog();
+                    CHECK_FAILURE(deferral->Complete());
+                };
+
+                return S_OK;
+            })
+        .Get(),
+    &m_downloadStartingToken));
+}
+
+// Update download progress
+void ScenarioCustomDownloadExperience::UpdateProgress(ICoreWebView2Download* download)
+{
+    // Register a handler for the `DownloadProgressSizeInBytesChanged` event.
+    CHECK_FAILURE(download->add_DownloadProgressSizeInBytesChanged(
+        Callback<ICoreWebView2DownloadProgressSizeInBytesChangedEventHandler>(
+            [this](ICoreWebView2Download* download, IUnknown* args) -> HRESULT {
+                // Here developer can update UI to show progress of a download using
+                // `download->get_DownloadProgressSizeInBytes` and
+                // `download->get_ExpectedDownloadSizeInBytes`.
+                return S_OK;
+            })
+        .Get(),
+    &m_downloadProgressSizeInBytesChangedToken));
+
+    // Register a handler for the `DownloadStateChanged` event.
+    CHECK_FAILURE(download->add_DownloadStateChanged(
+        Callback<ICoreWebView2DownloadStateChangedEventHandler>(
+            [this](ICoreWebView2Download* download,
+            ICoreWebView2DownloadStateChangedEventArgs* args) -> HRESULT {
+                COREWEBVIEW2_DOWNLOAD_STATE downloadState;
+                CHECK_FAILURE(download->get_State(&downloadState));
+                switch (downloadState)
+                {
+                case COREWEBVIEW2_DOWNLOAD_STATE_IN_PROGRESS:
+                    break;
+                case COREWEBVIEW2_DOWNLOAD_STATE_INTERRUPTED:
+                    // Here developer can take different actions based on `args->InterruptReason`.
+                    CompleteDownload(download);
+                    break;
+                case COREWEBVIEW2_DOWNLOAD_STATE_COMPLETED:
+                    CompleteDownload(download);
+                    break;
+                }
+                return S_OK;
+            })
+        .Get(),
+    &m_downloadStateChangedToken));
+}
+```
+## .NET/ WinRT
+```c#
+void DownloadStartingCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    if (_coreWebView2Settings == null)
+    {
+        _coreWebView2Settings = webView.CoreWebView2.Settings;
+    }
+    _coreWebView2Settings.ShouldDisplayDefaultDownloadDialog = false;
+
+    webView.CoreWebView2.DownloadStarting += delegate (object sender, CoreWebView2DownloadStartingEventArgs args)
+    {
+        CoreWebView2Deferral deferral = args.GetDeferral();
```

Comment:
Please add similar comments from my suggestions in the COM sample to here as well